OpenDreamRuntime/Procs/DMOpcodeHandlers.cs

```diff
@@ -1618,6 +1618,166 @@ private static void HandleSuffixPronoun(ref StringBuilder formattedString, ReadO
+                case DreamValue.DreamValueType.DreamObject when value == DreamValue.Null: state.Push(new DreamValue(MathF.Atan(0.0f))); break;
+                default: throw new Exception("Invalid argument type for arctan");
+            }
+
+            return ProcStatus.Continue;
+        }
+
+        public static ProcStatus Arctan2(DMProcState state) {
+            DreamValue y = state.Pop();
+            DreamValue x = state.Pop();
+
+            var yValue = 0.0f;
+
+            switch (y.Type) {
+                case DreamValue.DreamValueType.Float: yValue = y.MustGetValueAsFloat(); break;
+                case DreamValue.DreamValueType.DreamObject when y == DreamValue.Null: yValue = 0.0f; break;
+                default: throw new Exception("Invalid argument type for arctan2");
+            }
+
+            switch (x.Type) {
+                case DreamValue.DreamValueType.Float: state.Push(new DreamValue((float)(Math.Atan2(yValue, x.MustGetValueAsFloat()) / Math.PI * 180))); break;
+                case DreamValue.DreamValueType.DreamObject when x == DreamValue.Null: state.Push(new DreamValue(MathF.Atan2(yValue, 0.0f))); break;
+                default: throw new Exception("Invalid argument type for arctan2");
+            }
+
+            return ProcStatus.Continue;
+        }
+
+        public static ProcStatus Sqrt(DMProcState state) {
+            DreamValue value = state.Pop();
+
+            switch (value.Type) {
+                case DreamValue.DreamValueType.Float: state.Push(new DreamValue(MathF.Sqrt(value.MustGetValueAsFloat()))); break;
+                case DreamValue.DreamValueType.DreamObject when value == DreamValue.Null: state.Push(new DreamValue(MathF.Sqrt(0.0f))); break;
+                default: throw new Exception("Invalid argument type for sqrt");
+            }
+
+            return ProcStatus.Continue;
+        }
+
+        public static ProcStatus Log(DMProcState state) {
+            DreamValue baseValue = state.Pop();
+            DreamValue value = state.Pop();
+
+            float baseValueFloat = 0.0f;
+
+            switch (baseValue.Type) {
+                case DreamValue.DreamValueType.Float: baseValueFloat = baseValue.MustGetValueAsFloat(); break;
+                case DreamValue.DreamValueType.DreamObject when baseValue == DreamValue.Null: baseValueFloat = 0.0f; break;
+                default: throw new Exception("Invalid argument type for log");
+            }
+
+            switch (value.Type) {
+                case DreamValue.DreamValueType.Float: state.Push(new DreamValue(MathF.Log(value.MustGetValueAsFloat(), baseValueFloat))); break;
+                case DreamValue.DreamValueType.DreamObject when value == DreamValue.Null: state.Push(new DreamValue(MathF.Log(0.0f, baseValueFloat))); break;
+                default: throw new Exception("Invalid argument type for log");
+            }
+
```

Comment:
`UnsafeGetValueAsFloat()`
```suggestion
            state.Push(new DreamValue(MathF.Log(value.UnsafeGetValueAsFloat(), baseValue.UnsafeGetValueAsFloat())));
```